internal/scaffold/cmd.go

```diff
@@ -149,7 +150,26 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = serveCRMetrics(cfg); err != nil {
+	// Add the Metrics Service
+	addMetrics(ctx, cfg, namespace)
```

Comment:
It looks like `addMetrics()` is a small wrapper around `serveCRMetrics()` that does some extra logging and error handling, right?

IMO, another function in the `main.go` template for this is a bit much, and I think we could add the extra error handling in line in the body of the `main` function.

If I understand correctly, it seems like this PR could be reduced to:
```diff
        if err = serveCRMetrics(cfg); err != nil {
-               log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
+               if errors.Is(err, k8sutil.ErrRunLocal) {
+                       log.Info("Skipping CR metrics server creation; not running in a cluster.")
+               } else {
+                       log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
+               }
        }
```

Reply:
The logs/msgs came from more than one point in the original code (See [serveCRMetrics](https://github.com/operator-framework/operator-sdk/blob/bb234f88eb6815840c288bda70b0c481fdbf1ce2/internal/scaffold/cmd.go#L152), [CreateMetricsService](https://github.com/operator-framework/operator-sdk/blob/bb234f88eb6815840c288bda70b0c481fdbf1ce2/internal/scaffold/cmd.go#L162), [CreateServiceMonitors](https://github.com/operator-framework/operator-sdk/blob/bb234f88eb6815840c288bda70b0c481fdbf1ce2/internal/scaffold/cmd.go#L170)). The idea was centralized/encapsulate the logic to add the metrics and do its improvement. 

> {"level":"info","ts":1563485392.3758454,"logger":"cmd","msg":"Could not generate and serve custom resource metrics","error":"operator run mode forced to local"}
> {"level":"info","ts":1563485392.4013908,"logger":"cmd","msg":"Could not create metrics Service","error":"failed to initialize service object for metrics: OPERATOR_NAME must be set"}
> {"level":"info","ts":1563485392.4066937,"logger":"cmd","msg":"Could not create ServiceMonitor object","error":"no ServiceMonitor registered with the API"}

In this way, just the above suggestion still not attending the expectations.

Reply:
Ah I see now. The diff was confusing me :)